### DISCUSSION TITLE
How to create two way connection in my Prima model?

### DISCUSSION THREAD

**Discussion by Nethrenial** [prisma/prisma#12928] [Q&A]
I am creating a WhatsApp clone with nestjs and Prisma, as of now, I have a simple User model like this with an implicit friends relation (I need it to be two-way, so If I(User A) add someone as my friend(User B), he should also have me as a friend). Currently, my User model looks like this,

```prisma

model User {
  id              Int     @id @default(autoincrement())
  username        String  @unique() @db.VarChar(255)
  email           String  @unique() @db.VarChar(255)
  avatar          String? @db.VarChar(255)
  password        String  @default("1") @db.VarChar(255)
  friends         User[]  @relation("friends")
  friendsRelation User[]  @relation("friends")
  hashRT          String? @db.VarChar(255)

  @@map("users")
}

```
And I tried using this query to add an user as current user's friend,

```ts
  async addFriend(userId: number, friendId: number) {
    const user = await prisma.user.update({
      where: {
        id: userId,
      },
      data: {
        friends: {
          connect: {
            id: friendId,
          },
        },
      },
      include: {
        friends: true,
      },
    });
    return user;
  }
```

with this, when I query to get the user's friend list using the following query, I get the new friend in the result(so he was added successfully)

```ts
  async getFriends(userId: number) {
    const user = await prisma.user.findFirst({
      where: {
        id: userId,
      },
      include: {
        friends: true,
      },
    });
    return user.friends;
  }
```
So I'm confident that the friend was added successfully. However, when I do the same query for the new friend, his friends' array is empty. Meaning he doesn't know that he has a friend.
But I want the relationship to be two-way, so when I add someone as a friend(User B) of UserA, I need the current User A to be set as a friend. of User B as well.
I got it to work with two queries,

```ts
 async addFriend(userId: number, friendId: number) {
    const user = await prisma.user.update({
      where: {
        id: userId,
      },
      data: {
        friends: {
          connect: {
            id: friendId,
          },
        },
      },
      include: {
        friends: true,
      },
    });
    //also add user to friend's friends
    await prisma.user.update({
      where: {
        id: friendId,
      },
      data: {
        friends: {
          connect: {
            id: userId,
          },
        },
      },
      include: {
        friends: false,
        _count: false,
        friendsRelation: false,
      },
    });
    return user;
  }
```
Is this the correct way of doing this? Or is there any predefined way of handling this? I'm open to schema changes if they can help with this situation.

I'm using prisma version 3.12.0 and MySQL as the provider.

**Comment by pantharshit00** [comment]
Hello @Nethrenial, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. 

Yes, this will be the correct way. You can also alternatively query `friendsRelation` and merge it with friends. If you look at the underlying database table:
<img width="644" alt="image" src="https://user-images.githubusercontent.com/22195362/196602986-1228469e-d2a6-4c09-97f8-0870c4b0952a.png">

`A` side is `friendsRelation` and `B` side is `friends`. So in this case, if you query from side of user with ID of 1, it can only access data via `friends` side while user with ID 2 can access data from `friendsRelation` side. So you will need to add a row here with `1` to `2` in the link table to also get it using the `friends` property in the other user side. As I have mentioned you can also merge these two. I think you can also rename `friendsRelation` to `friendsAddedMe` to make it more clear.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @Nethrenial, I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Reply by RodrigoNovais** [reply]
@vladi-stevanovic I'm in a similar situation and would like to know if it's possible to achieve that same result in a single raw query

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
